File: tests/llvm/hardlift/benchmark/numba/overlay_blend_8.py
from numba import njit

@njit
def overlay_blend_8(base, active):
  output = []
  m = len(base)
  n = len(base[0])
  for i in range(m):
    curr_row = []
    for j in range(n):
      if base[i][j] >= 128:
        pixel = 2 * base[i][i] + base[i][j] - 2 * base[i][j] * base[i][j] / 255 - 255
      else:
        pixel = 2 * base[i][j] * base[i][j] / 255
      curr_row.append(pixel)
    output.append(curr_row)
  return output

File: tenspiler/dexter/cpp/for_synthesis/overlay_blend_8.cc
#include <vector>
using namespace std;

vector<vector<int>> overlay_blend_8(vector<vector<int>> base, vector<vector<int>> active)
{
    vector<vector<int>> out;
    int m = base.size();
    int n = base[0].size();
	for (int row = 0; row < m; row++) {
        vector<int> row_vec;
		for (int col = 0; col < n; col++) {
			int pixel;
			if (base[row][col] >= 16)
                pixel = 2 * base[row][col] + base[row][col] - 2 * base[row][col] * base[row][col] / 32 - 32;
			else
                pixel = 2 * base[row][col] * base[row][col] / 32;
			row_vec.push_back(pixel);
		}
		out.push_back(row_vec);
	}
	return out;
}
